module/Activity/src/Activity/Service/Activity.php

```diff
@@ -119,346 +342,347 @@ private function requestGEFLITST($activity, $user, $organ)
+        // proposal. We unset all `id`s after getting the diff to reduce the
+        // number of calls.
+        $diff = $this->array_diff_assoc_recursive($current, $proposal);
+        $this->recursiveUnsetKey($diff, 'id');
+
+        // Filter out all empty parts of the difference, if we an empty result
+        // nothing has changed on form submission.
+        if (empty($this->array_filter_recursive($diff))) {
             return false;
         }
 
-        // If the activity has not been approved yet the update proposal can be applied
-        return $activity->getStatus() !== ActivityModel::STATUS_APPROVED;
+        return true;
     }
 
     /**
-     * Apply a proposed activity update
+     * `array_diff_assoc` but recursively. Used to compare an update proposal of an activity
+     * to the original activity.
      *
-     * @param ActivityProposalModel $proposal
-     */
-    public function updateActivity(ActivityProposalModel $proposal)
-    {
-        $old = $proposal->getOld();
-        $new = $proposal->getNew();
-        $this->copyActivity($old, $new);
-        $em = $this->getServiceManager()->get('Doctrine\ORM\EntityManager');
-
-        $em->remove($proposal);//Proposal is no longer needed.
-        $em->remove($new);
-        $em->flush();
-    }
-
-    /**
-     * Copies all relevant activity attributes from $new to $old
+     * Adapted from https://www.php.net/manual/en/function.array-diff-assoc.php#usernotes.
      *
-     * @param ActivityModel $old
-     * @param ActivityModel $new
+     * @param array $array1
+     * @param array $array2
+     * @return bool
      */
-    protected function copyActivity(ActivityModel $old, ActivityModel $new)
-    {
-        $old->setName($new->getName());
-        $old->setNameEn($new->getNameEn());
-        $old->setBeginTime($new->getBeginTime());
-        $old->setEndTime($new->getEndTime());
-        $old->setSubscriptionDeadline($new->getSubscriptionDeadline());
-        $old->setLocation($new->getLocation());
-        $old->setLocationEn($new->getLocationEn());
-        $old->setCosts($new->getCosts());
-        $old->setCostsEn($new->getCostsEn());
-        $old->setDescription($new->getDescription());
-        $old->setDescriptionEn($new->getDescriptionEn());
-        $old->setCreator($new->getCreator());
-        $old->setCanSignUp($new->getCanSignUp());
-        $old->setOnlyGEWIS($new->getOnlyGEWIS());
-    }
+    protected function array_diff_assoc_recursive(array $array1, array $array2) {
+        $difference = [];
+
+        foreach ($array1 as $key => $value) {
+            if (is_array($value)) {
+                if (!array_key_exists($key, $array2) || !is_array($array2[$key])) {
+                    $difference[$key] = $value;
+                } else {
+                    $newDifference = $this->array_diff_assoc_recursive($value, $array2[$key]);
+
+                    if (!empty($new_diff)) {
```

Comment:
I think new_diff is undefined. Shoudln't this be newDifference?
This seems like a bug,